packages/storage/src/implementation/metadata.ts

```diff
@@ -161,57 +167,60 @@ export function downloadUrlFromResourceString(
     // through list, so we don't want to throw an Error.
     return null;
   }
-  let tokens: string = obj['downloadTokens'] as string;
+  const tokens: string = obj['downloadTokens'] as string;
   if (tokens.length === 0) {
     return null;
   }
-  let encode = encodeURIComponent;
-  let tokensList = tokens.split(',');
-  let urls = tokensList.map(function(token: string): string {
-    let bucket: string = metadata['bucket'] as string;
-    let path: string = metadata['fullPath'] as string;
-    let urlPart = '/b/' + encode(bucket) + '/o/' + encode(path);
-    let base = UrlUtils.makeUrl(urlPart);
-    let queryString = UrlUtils.makeQueryString({
-      alt: 'media',
-      token: token
-    });
-    return base + queryString;
-  });
+  const encode = encodeURIComponent;
+  const tokensList = tokens.split(',');
+  const urls = tokensList.map(
+    (token: string): string => {
+      const bucket: string = metadata['bucket'] as string;
+      const path: string = metadata['fullPath'] as string;
+      const urlPart = '/b/' + encode(bucket) + '/o/' + encode(path);
+      const base = UrlUtils.makeUrl(urlPart);
+      const queryString = UrlUtils.makeQueryString({
+        alt: 'media',
+        token
+      });
+      return base + queryString;
+    }
+  );
   return urls[0];
 }
 
 export function toResourceString(
   metadata: Metadata,
   mappings: Mappings
 ): string {
-  let resource: {
-    [prop: string]: any;
+  const resource: {
+    [prop: string]: unknown;
   } = {};
-  let len = mappings.length;
+  const len = mappings.length;
   for (let i = 0; i < len; i++) {
-    let mapping = mappings[i];
+    const mapping = mappings[i];
     if (mapping.writable) {
       resource[mapping.server] = metadata[mapping.local];
     }
   }
   return JSON.stringify(resource);
 }
 
-export function metadataValidator(p: any) {
-  let validType = p && type.isObject(p);
-  if (!validType) {
+export function metadataValidator(p: unknown): void {
+  if (!(p && type.isObject(p))) {
     throw 'Expected Metadata object.';
   }
-  for (let key in p) {
-    let val = p[key];
-    if (key === 'customMetadata') {
-      if (!type.isObject(val)) {
-        throw 'Expected object for \'customMetadata\' mapping.';
-      }
-    } else {
-      if (type.isNonNullObject(val)) {
-        throw "Mapping for '" + key + "' cannot be an object.";
+  for (const key in p) {
+    if (contains(p, key)) {
```

Comment:
added the hasownproperty guard. Is it correct? @schmidt-sebastian

Reply:
I think so. I inline `contains` in my follow-up PR.